moodlecli/main.py

```diff
@@ -275,6 +277,8 @@ def export_bulk(input_csv, bucket_name, directory, data_type):
             aws.put_json_data(grade_data, bucket_name, key)
         elif data_type == 'users':
             user_data = moodle.get_users_by_course(id)
+            uuid_data = moodle.get_user_uuids([])
```

Comment:
It just occurred to me: Since we're using the approach of merging UUIDs into the users JSON data, merging this PR will break the corresponding commands when run against environments where the plugin support for the endpoint does not yet exist. I believe the only active dependency we have on that is the metrics analyzer which luckily uses a Docker image and also [pins the version](https://github.com/openstax/raise-moodle-spikes/blob/b8621988a29c580e0c39bf8270ef775455e34c67/reporting/raise-metrics-analyzer/requirements.txt#L2) of the CLI, so that should continue running okay. I'm just calling out the general issue in case anyone else can think of where this might bite us until we get the plugin deployed in staging / prod. If we wanted to provide backwards compat, we could catch the error and use `[]` as the `uuid_data` value. 

Thoughts @mjoss17 , @P-Gill97 ?

Reply:
Hmm, I can toss in the catch for a lack of endpoint right now - good idea. 

I think we should be ok CLI-wise. We don't have any automated calls to `export-bulk` or `export-bulk-users` right now. Those are the only two commands that use the endpoint in question. It just would not be a good idea to populate any sort of enclave again until we get the plugin set. So Iris might have to make do with the data that's there right now.

Reply:
@mjoss17 - Oh, we update the S3 data using `export-bulk` every night in scheduled [the argo workflow](https://github.com/openstax/raise-moodle-spikes/blob/b8621988a29c580e0c39bf8270ef775455e34c67/reporting/workflows/raise-data-daily.yaml#L22). But that use case is okay (for now) since the version is pinned. 

But, yeah, to be safe it seems prudent to make the CLI command backwards compatible for now.

Reply:
@mjoss17 - Also, just as a suggestion for implementation: You may want to create a utils function to query the UUIDs so the logic of error handling all that is written once and used in both places in `main.py`.

Reply:
Ah, okay. I see in the README there thats similar / what you meant by daily metrics analyzer. 

Quick question - What is the significance of having this argo workflow in the spikes repo? (Sometimes I get stuck imagining everything in spikes is just a demo) Why didn't we put the workflow raise-moodlecli for example?

Reply:
The only reason it's in spikes is that I wrote it as part of a spike. At some point probably I'll move the entire `reporting` directory there to a `raise-reporting` repo or something (I don't think it belongs in the CLI repo), but I wanted to hold off until I felt like it was clearer what all would need to go in such a repo. spikes is basically a temporary buffer (and it's simple enough to carry commits over to the new repo when we create it, so it's not big deal in terms of losing git history. I did that [previously](https://github.com/openstax/k12/issues/30#issuecomment-1005842596) when setting up the `local_raisecli` plugin).

Reply:
To [this question](https://github.com/openstax/raise-moodlecli/pull/16#issuecomment-1269009463): Ah, I see, you are trying to filter on the specific error so any unexpected error doesn't get hidden away, which seems like the right implementation. Could you perhaps get the coverage by writing a unit test on `maybe_user_data` where you pass a mock `moodle_client` that will throw the error? That way you don't need to mess with the fixtures at all :man_shrugging: .